Civi/Api4/Service/Spec/SpecFormatter.php

```diff
@@ -58,7 +59,8 @@ public static function arrayToField(array $data, $entity) {
       $field = new FieldSpec($name, $entity, $dataTypeName);
       $field->setType('Field');
       $field->setColumnName($name);
-      $field->setRequired(!empty($data['required']));
+      $field->setNullable(empty($data['required']));
```

Comment:
Is this an "approximation" for now? For example I don't think activity date or status are really "nullable".

Reply:
Both of those fields are nullable in our schema. I think you could make a very good argument that they shouldn't be. Fixing it on the schema level can be a pain though because you have to deal with any existing data with null values. A quicker fix would be to set nullable=false just at the APIv4 layer for those fields (via the `ActivitySpecProvider`). I can do that as a followup.

Reply:
Ok. My only other comment then was the one below about custom date fields. Otherwise looked good.